app/src/test/java/com/linagora/calendar/app/restapi/routes/WebsocketRouteTest.java

```diff
@@ -0,0 +1,412 @@
+
+        webSocket.send("""
+            {
+                "register": ["%s"]
+            }
+            """.formatted(registerCalendarURL));
+
+        String json = messages.poll(5, TimeUnit.SECONDS);
+        assertThat(json).isNotNull();
+
+        assertThatJson(json)
+            .isEqualTo("""
+                {
+                    "registered": ["%s"]
+                }
+                """.formatted(registerCalendarURL));
+    }
+
+    @Test
+    void websocketShouldUnregisterSuccessfully() throws Exception {
+        String calendarUri = CalendarURL.from(bob.id()).asUri().toString();
+
+        String ticket = generateTicket(bob);
+        BlockingQueue<String> messages = new LinkedBlockingQueue<>();
+        webSocket = connectWebSocket(restApiPort, ticket, messages);
+
+        webSocket.send("""
+            {
+                "unregister": ["%s"]
+            }
+            """.formatted(calendarUri));
+
+        String unregJson = messages.poll(5, TimeUnit.SECONDS);
+        assertThat(unregJson).isNotNull();
+        assertThatJson(unregJson)
+            .isEqualTo("""
+                {
+                    "unregistered": ["%s"]
+                }
+                """.formatted(calendarUri));
+    }
+
+    @Test
+    void websocketShouldHandleMixedRegisterAndUnregister() throws Exception {
+        // Given: A & C belong to bob (valid)
+        String calendarA = CalendarURL.from(bob.id()).asUri().toString();
+        CalDavClient.NewCalendar publicCalendar = new CalDavClient.NewCalendar(UUID.randomUUID().toString(),
+            "Public Calendar", "#123456", "Public calendar for testing");
+        calDavClient.createNewCalendar(bob.username(), bob.id(), publicCalendar).block();
+        String calendarC = new CalendarURL(bob.id(), new OpenPaaSId(publicCalendar.id())).asUri().toString();
+
+        // B does not exist → should fail register
+        String calendarB = new CalendarURL(bob.id(), new OpenPaaSId(UUID.randomUUID().toString()))
+            .asUri().toString();
+
+        String ticket = generateTicket(bob);
+        BlockingQueue<String> messages = new LinkedBlockingQueue<>();
+        webSocket = connectWebSocket(restApiPort, ticket, messages);
+
+        // When: send register A+B and unregister C in same payload
+        webSocket.send("""
+            {
+                "register": ["%s", "%s"],
+                "unregister": ["%s"]
+            }
+            """.formatted(calendarA, calendarB, calendarC));
+
+        String json = messages.poll(5, TimeUnit.SECONDS);
+        assertThat(json).isNotNull();
+
+        // Then: A registered, B notRegistered, C unregistered
+        assertThatJson(json)
+            .isEqualTo("""
+                {
+                  "registered": ["%s"],
+                  "notRegistered": ["%s"],
+                  "unregistered": ["%s"]
+                }
+                """.formatted(calendarA, calendarB, calendarC));
+    }
+
```

Comment:
What happens if I unregister a caledarURL I did not register before?

Reply:
Added a test verifying idempotency of implicit register/unregister.